### DISCUSSION TITLE
Turn python function into sympy expression

### DISCUSSION THREAD

**Discussion by abkein** [sympy/sympy#24001] [General]
For example, I have the function, that accepts only floats:
```
def do_smth(a: float, b: float) -> float:
    return a + b
```
This is not necessarily a summation, it can be any function (for example, a function after `autowrap`) - what matters is that it does not accept `Symbol` as arguments

How can I turn it into sympy expression, which only evaluated when the numbers are substituted?

Something like this:
```
from sympy.abc import x,y
wf = wrap2sympy(do_smth, args=(x,y))
wf.subs(x, 5).subs(y, 0.5).evalf()   # 5.5 should be returned
```
Or this
```
from sympy.utilities.autowrap import autowrap
from sympy.abc import a, b, x, y

expr = (a + b)**a
expr_compiled = autowrap(expr, args=(a, b), tempdir="somedir")
wf = wrap2sympy(expr_compiled, args=(x,y))
wf.subs(x, 2).subs(y, 2).evalf()   # 16 should be returned
```
Function `wrap2sympy` does not exists and I'm asking for its implementation

**Comment by abkein** [comment]
It's a duplicate of [this (stackoverflow)](https://stackoverflow.com/questions/72443207/how-to-add-a-callable-function-to-a-sympy-expression/72450342#72450342)

**Reply by oscarbenjamin** [reply]
I think that SO answer could be something that is good to add to SymPy.

**Comment by asmeurer** [comment]
If it is after autowrap that is going to be more difficult, because an autowrapped function is compiled code, so you'd have to extract the expression from the compiled executable. There might be tools that could do this if you look, but it's probably not something you'd want to write by hand. 

If it's a Python function you could try to extract the source using tools from the `inspect` module and convert them using `sympify` or `parse_expr`, assuming it involves something like NumPy functions.  If it's just arithmetic operations you can just pass symbols as arguments to the function.

**Reply by abkein** [reply]
I don't want to extract them, just call
